src/mantarray_desktop_app/mc_comm.py

```diff
@@ -30,21 +80,117 @@ def __init__(self, *args: Any, **kwargs: Any):
             self._board_queues
         )
 
+    def _setup_before_loop(self) -> None:
+        msg = {
+            "communication_type": "log",
+            "message": f'Microcontroller Communication Process initiated at {datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S.%f")}',
+        }
+        to_main_queue = self._board_queues[0][1]
+        if not self._suppress_setup_communication_to_main:
+            to_main_queue.put(msg)
```

Comment:
note for future: should only use `put` with a `timeout` kwarg or `put_nowait`, to avoid risk of hanging